Dockerfile.e2e-tests

```
@@ -0,0 +1,23 @@
+# By default we pin to amd64 sha. Use make docker to automatically adjust for arm64 versions.
+ARG BASE_DOCKER_SHA="14d68ca3d69fceaa6224250c83d81d935c053fb13594c811038c461194599973"
+
+# Taking a non-alpine image for e2e tests so that cgo can be enabled
+FROM golang:1.17 as builder
+
+WORKDIR $GOPATH/src/github.com/thanos-io/thanos
+# Change in the docker context invalidates the cache so to leverage docker
+# layer caching, moving update and installing apk packages above COPY cmd
+# More info https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache
+
+COPY . $GOPATH/src/github.com/thanos-io/thanos
+
+RUN git update-index --refresh; make build
```

Should this be `build-e2e`? Otherwise we're building binary without the race detector, right?

```suggestion
RUN git update-index --refresh; make build-e2e
```

---

Tested locally and this works alright!